web/views.py

```
@@ -948,6 +975,178 @@ def student_progress(request, enrollment_id):
+
+    weeks_since_first_session = 1
+    if all_completed_sessions:
+        first_session = min(all_completed_sessions, key=lambda s: s.start_time)
+        weeks_since_first_session = max(1, (now - first_session.start_time).days / 7)
+
+    avg_sessions_per_week = round(len(all_completed_sessions) / weeks_since_first_session, 1)
+
+    # Completion pace
+    completion_pace = "N/A"
+    if total_courses > 0:
+        if courses_completed > 0:
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

Combine `if` statements using `and`

(SIM102)